packages/core/src/tenants/SystemContext.ts

```diff
@@ -44,6 +45,13 @@ export default class SystemContext {
           protectedAppConfigProviderDataGuard
         );
       })(),
+      (async () => {
+        this.protectedAppHostnameProviderConfig = await this.loadConfig(
+          pool,
+          CloudflareKey.ProtectedAppHostnameProvider,
+          hostnameProviderDataGuard
+        );
+      })(),
```

Comment:
merge with the existing hostNameProvirer loader?

Reply:
Although the types are the same, but they are 2 different configs, in other words, the 2 "custom domain" features are independent, and the config details (zoneId) are diffrent.